crypto/rand/drbg_lib.c

```diff
@@ -310,12 +382,20 @@ int rand_drbg_restart(RAND_DRBG *drbg,
     }
 
     /* repair error state */
-    if (drbg->state == DRBG_ERROR)
+    if (drbg->state == DRBG_ERROR) {
         RAND_DRBG_uninstantiate(drbg);
+        /* The drbg->ctr member needs to be reinitialized before reinstantiation */
+        RAND_DRBG_set(drbg, drbg->nid, drbg->flags);
+    }
 
     /* repair uninitialized state */
     if (drbg->state == DRBG_UNINITIALISED) {
-        drbg_setup(drbg, NULL);
+        /* reset reseed time */
+        drbg->reseed_time = 1;
```

Comment:
Does that need an `if (drbg->reseed_time != 0)`?

Reply:
Yes, indeed!